### ISSUE TITLE
Inform developer when signing is the reason the app can't run on iOS

### ISSUE THREAD

**Issue by sethladd** [flutter/flutter#2952]
Unfortunately, it's not clear what action I can take, as a developer. Can we give more guidance about what I should do now?

I'm trying to run the gallery app on my iPad, and I got this:

```
~/Code/flutter/examples/material_gallery[master*] $ flutter -d ipad run
Running 'pub get' in /Users/sethladd/Code/flutter/examples/material_gallery/...
Uploading .generated package contents... DONE.
Installing '(null)'
 - CreatingStagingDirectory (5%)
 - ExtractingPackage (15%)
 - InspectingPackage (20%)
 - Error occurred: PackageInspectionFailed
Running lib/main.dart on iPad...
Unzipping Xcode project to local directory...
Xcode project created in /Users/sethladd/Code/flutter/examples/material_gallery/ios/.
[....] Waiting for iOS device to be connected
[....] Using J81AP 'iPad' (b0f7081e6f02d7e3b573bf6afcb64e61ed8b86bb).
------ Install phase ------
[  0%] Found J81AP 'iPad' (b0f7081e6f02d7e3b573bf6afcb64e61ed8b86bb) connected through USB, beginning install
[  5%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/META-INF/ to device
[  5%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/META-INF/com.apple.ZipMetadata.plist to device
[  6%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/_CodeSignature/ to device
[  6%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/_CodeSignature/CodeResources to device
[  7%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon29x29@2x.png to device
[  8%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon29x29@2x~ipad.png to device
[  8%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon29x29@3x.png to device
[  9%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon29x29~ipad.png to device
[ 10%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon40x40@2x.png to device
[ 10%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon40x40@2x~ipad.png to device
[ 11%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon40x40@3x.png to device
[ 12%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon40x40~ipad.png to device
[ 12%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon60x60@2x.png to device
[ 13%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon60x60@3x.png to device
[ 14%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon76x76@2x~ipad.png to device
[ 14%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon76x76~ipad.png to device
[ 15%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/AppIcon83.5x83.5@2x~ipad.png to device
[ 16%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/embedded.mobileprovision to device
[ 17%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/Frameworks/ to device
[ 17%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/Frameworks/FlutterApplication.framework/ to device
[ 18%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/Frameworks/FlutterApplication.framework/_CodeSignature/ to device
[ 19%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/Frameworks/FlutterApplication.framework/_CodeSignature/CodeResources to device
[ 19%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/Frameworks/FlutterApplication.framework/app.flx to device
[ 24%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/Frameworks/FlutterApplication.framework/FlutterApplication to device
[ 33%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/Frameworks/FlutterApplication.framework/Info.plist to device
[ 33%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/icudtl.dat to device
[ 37%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/Info.plist to device
[ 37%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/LaunchScreen.storyboardc/ to device
[ 38%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/LaunchScreen.storyboardc/01J-lp-oVM-view-Ze5-6b-2t3.nib to device
[ 39%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/LaunchScreen.storyboardc/Info.plist to device
[ 39%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/LaunchScreen.storyboardc/UIViewController-01J-lp-oVM.nib to device
[ 40%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/PkgInfo to device
[ 40%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/Runner to device
[ 49%] Copying /Users/sethladd/Code/flutter/examples/material_gallery/ios/.generated/build/Release-iphoneos/Runner.app/ServiceDefinitions.json to device
[ 52%] CreatingStagingDirectory
[ 57%] ExtractingPackage
[ 60%] InspectingPackage
[ 60%] TakingInstallLock
[ 65%] PreflightingApplication
[ 65%] InstallingEmbeddedProfile
[ 70%] VerifyingApplication
2016-03-28 10:56:42.066 ios-deploy[34409:491933] [ !! ] Error 0xe8008015: ·ú7zˇˇ AMDeviceSecureInstallApplication(0, device, url, options, install_callback, 0)
Could not install ios/.generated/build/Release-iphoneos/Runner.app on b0f7081e6f02d7e3b573bf6afcb64e61ed8b86bb.
Error running application on iPad.
```

**Comment by sethladd** [comment]
cc @chinmaygarde @tvolkert

**Comment by sethladd** [comment]
Output of `flutter doctor`

```
~/Code/flutter/examples/material_gallery[master*] $ flutter doctor
[✓] Flutter (on Mac OS, channel master)
    • Flutter at /Users/sethladd/Code/flutter
    • Framework revision cc9b140 (43 minutes ago), engine revision f790903

[✓] Android toolchain - develop for Android devices (Android SDK 23.0.2)
    • Android SDK at /Users/sethladd/homebrew/opt/android-sdk
    • Platform android-23, build-tools 23.0.2
    • Java(TM) SE Runtime Environment (build 1.8.0_72-b15)

[✓] iOS toolchain - develop for iOS devices (Xcode 7.2)
    • XCode at /Applications/Xcode.app/Contents/Developer
    • Xcode 7.2, Build version 7C68

[✓] Atom - a lightweight development environment for Flutter
    • Atom installed; Flutter plugin version 0.2.1
```

**Comment by chinmaygarde** [comment]
Thats a code signing issue. I am going to try to add doctor checks for this.

**Comment by sethladd** [comment]
Even better, can you print out something like "Sorry, we tried to run code signing, but it failed. Please read this doc for possible ways to fix." ?

**Comment by sethladd** [comment]
Hm, I was able to run a sample app from Xcode on this tablet (after I went into my tablet's setting and trusted my account), but I'm still getting the same `0xe8008015` error when I try to `flutter -d ipad run`.

**Comment by sethladd** [comment]
I opened my test app in XCode, and ran it once via XCode's tooling. I had to choose a team, but once I did that, and ensured I had the right key chosen, I was able to run the app via Xcode. I then tried the command-line tools and was able to `flutter -d ipad run`.

As we discussed, perhaps we can add a check during `run` that detects when security/signing is the reasons the `run` command might fail, and then notify the user (and maybe even fail fast?)
